### ISSUE TITLE
libicsneolegacy.so: undefined symbol: _ZN6icsneo10NeoVIFIRE218SUPPORTED_NETWORKSE

### ISSUE THREAD

**Issue by drebbe-intrepid** [intrepidcs/libicsneo#1]
I believe this to be a gcc linker build order issue but not 100%. This was first reported here: https://github.com/intrepidcs/python_ics/issues/34 and reproduced by me.

Built using the following:
```
mkdir build && cd build
cmake ..
make
```

```
drebbe@localhost dev]$ python3
Python 3.6.7 (default, Nov 23 2018, 12:11:28) 
[GCC 8.2.1 20181105 (Red Hat 8.2.1-5)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ics
>>> ics.override_library_name("/home/drebbe/dev/libicsneo/build/libicsneolegacy.so")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ics.RuntimeError: Error: override_library_name(): Failed to open library '/home/drebbe/dev/libicsneo/build/libicsneolegacy.so': /home/drebbe/dev/libicsneo/build/libicsneolegacy.so: undefined symbol: _ZN6icsneo10NeoVIFIRE218SUPPORTED_NETWORKSE
>>> 

```

```
[drebbe@localhost build]$ nm -D libicsneolegacy.so | grep SUPPORTED_NETWORK
                 U _ZN6icsneo10NeoOBD2PRO18SUPPORTED_NETWORKSE
                 U _ZN6icsneo10NeoOBD2SIM18SUPPORTED_NETWORKSE
                 U _ZN6icsneo10NeoVIFIRE218SUPPORTED_NETWORKSE
                 U _ZN6icsneo11ValueCAN4_118SUPPORTED_NETWORKSE
                 U _ZN6icsneo11ValueCAN4_218SUPPORTED_NETWORKSE
                 U _ZN6icsneo11ValueCAN4_418SUPPORTED_NETWORKSE
                 U _ZN6icsneo13ValueCAN4_2EL18SUPPORTED_NETWORKSE
                 U _ZN6icsneo8RADStar218SUPPORTED_NETWORKSE
                 U _ZN6icsneo9NeoVIFIRE18SUPPORTED_NETWORKSE
                 U _ZN6icsneo9ValueCAN318SUPPORTED_NETWORKSE
```

Computer information:

```
[drebbe@localhost libicsneo]$ hostnamectl
   Static hostname: localhost.localdomain
         Icon name: computer-vm
           Chassis: vm
        Machine ID: ebc87001086749709289e8bd5fe688b7
           Boot ID: 420a4dda4f724c028fa5c8de6c5a5e8d
    Virtualization: oracle
  Operating System: Fedora 28 (Workstation Edition)
       CPE OS Name: cpe:/o:fedoraproject:fedora:28
            Kernel: Linux 4.19.13-200.fc28.x86_64
      Architecture: x86-64

```
Package information:
```
[drebbe@localhost dev]$ rpm -qa --last | head -n37
libconfuse-devel-3.2.2-1.fc28.x86_64          Thu 10 Jan 2019 05:02:22 PM EST
libconfuse-3.2.2-1.fc28.x86_64                Thu 10 Jan 2019 05:02:22 PM EST
boost-devel-1.66.0-8.fc28.x86_64              Thu 10 Jan 2019 05:01:20 PM EST
libquadmath-devel-8.2.1-6.fc28.x86_64         Thu 10 Jan 2019 05:01:11 PM EST
libicu-devel-60.2-2.fc28.x86_64               Thu 10 Jan 2019 05:01:11 PM EST
boost-numpy-1.66.0-8.fc28.x86_64              Thu 10 Jan 2019 05:01:11 PM EST
boost-1.66.0-8.fc28.x86_64                    Thu 10 Jan 2019 05:01:11 PM EST
python2-numpy-1.14.5-1.fc28.x86_64            Thu 10 Jan 2019 05:01:10 PM EST
python2-nose-1.3.7-19.fc28.noarch             Thu 10 Jan 2019 05:01:09 PM EST
openblas-serial-0.3.3-2.fc28.x86_64           Thu 10 Jan 2019 05:01:09 PM EST
openblas-threads-0.3.3-2.fc28.x86_64          Thu 10 Jan 2019 05:01:08 PM EST
boost-container-1.66.0-8.fc28.x86_64          Thu 10 Jan 2019 05:01:08 PM EST
boost-math-1.66.0-8.fc28.x86_64               Thu 10 Jan 2019 05:01:07 PM EST
boost-random-1.66.0-8.fc28.x86_64             Thu 10 Jan 2019 05:01:06 PM EST
boost-program-options-1.66.0-8.fc28.x86_64    Thu 10 Jan 2019 05:01:06 PM EST
boost-serialization-1.66.0-8.fc28.x86_64      Thu 10 Jan 2019 05:01:05 PM EST
boost-signals-1.66.0-8.fc28.x86_64            Thu 10 Jan 2019 05:01:04 PM EST
boost-type_erasure-1.66.0-8.fc28.x86_64       Thu 10 Jan 2019 05:01:03 PM EST
boost-stacktrace-1.66.0-8.fc28.x86_64         Thu 10 Jan 2019 05:01:03 PM EST
openblas-0.3.3-2.fc28.x86_64                  Thu 10 Jan 2019 05:01:02 PM EST
boost-wave-1.66.0-8.fc28.x86_64               Thu 10 Jan 2019 05:01:02 PM EST
boost-graph-1.66.0-8.fc28.x86_64              Thu 10 Jan 2019 05:01:02 PM EST
boost-fiber-1.66.0-8.fc28.x86_64              Thu 10 Jan 2019 05:01:01 PM EST
boost-coroutine-1.66.0-8.fc28.x86_64          Thu 10 Jan 2019 05:01:01 PM EST
boost-timer-1.66.0-8.fc28.x86_64              Thu 10 Jan 2019 05:01:00 PM EST
boost-test-1.66.0-8.fc28.x86_64               Thu 10 Jan 2019 05:01:00 PM EST
boost-python2-1.66.0-8.fc28.x86_64            Thu 10 Jan 2019 05:01:00 PM EST
boost-log-1.66.0-8.fc28.x86_64                Thu 10 Jan 2019 05:01:00 PM EST
boost-context-1.66.0-8.fc28.x86_64            Thu 10 Jan 2019 05:01:00 PM EST
boost-atomic-1.66.0-8.fc28.x86_64             Thu 10 Jan 2019 05:01:00 PM EST
boost-regex-1.66.0-8.fc28.x86_64              Thu 10 Jan 2019 05:00:59 PM EST
boost-jam-1.66.0-8.fc28.x86_64                Thu 10 Jan 2019 04:56:50 PM EST
boost-build-1.66.0-8.fc28.noarch              Thu 10 Jan 2019 04:56:50 PM EST
libusbx-devel-1.0.22-1.fc28.x86_64            Thu 10 Jan 2019 04:56:11 PM EST
libusb-devel-0.1.5-12.fc28.x86_64             Thu 10 Jan 2019 04:56:11 PM EST
cmake-3.11.2-1.fc28.x86_64                    Thu 10 Jan 2019 04:53:06 PM EST
cmake-data-3.11.2-1.fc28.noarch               Thu 10 Jan 2019 04:53:05 PM EST
```

python_ics library can be installed by the following command:

```
$ pip3 install git+https://github.com/intrepidcs/python_ics.git@v802
```

**Comment by hollinsky-intrepid** [comment]
Resolved as of c313801, the legacy API now loads into python_ics under Linux correctly.

**Comment by XhantorzD** [comment]
I'm using the NeoVI FIRE/Red and I seem to have issues setting the baudrate for CAN lines. Is there a specific firmware that is supported? Thanks.

```console
Running libicsneo v0.1.0 c31380
Supported devices:
        neoOBD2 PRO
        neoOBD2 SIM
        neoVI FIRE
        neoVI FIRE 2
        neoVI ION
        neoVI PLASMA
        RADStar 2
        RADSupermoon
        ValueCAN 3
        ValueCAN 4-1
        ValueCAN 4-2
        ValueCAN 4-2EL
        ValueCAN 4-4
        VividCAN

Finding devices... OK, 1 device found
        neoVI FIRE - XXXXX @ Handle 1

Connecting to neoVI FIRE XXXXX ... OK
        Getting HSCAN Baudrate... FAIL
        Setting HSCAN to operate at 125kbit/s... FAIL
        Getting HSCAN Baudrate... (expected to be unchanged) FAIL
        Getting HSCANFD Baudrate... FAIL
        Setting HSCANFD to operate at 8Mbit/s... FAIL
        Getting HSCANFD Baudrate... (expected to be unchanged) FAIL
        Setting settings temporarily... FAIL
        Getting HSCAN Baudrate... FAIL
        Getting HSCANFD Baudrate... FAIL
        Setting settings permanently... FAIL

        Going online... OK
        Checking online status... OK
        Streaming messages in for 3 seconds...
                Got 0 messages while polling
        Transmitting an extended CAN FD frame... FAIL
        Transmitting an ethernet frame on OP (BR) Ethernet 2... FAIL
        Going offline... OK
        Setting default settings... FAIL
        Disconnecting... OK
```
